packages/cli/src/index.js

```diff
@@ -31,69 +32,83 @@ import * as typeCheckCommand from './commands/type-check'
+// 3. By traversing directories upwards for the first `redwood.toml`
+//
+// ## Examples
+//
+// ```
+// yarn rw info --cwd /path/to/project
+// RWJS_CWD=/path/to/project yarn rw info
+//
+// # In this case, `--cwd` wins out
+// RWJS_CWD=/path/to/project yarn rw info --cwd /path/to/other/project
+//
+// # Here `findup` traverses upwards.
+// cd api
+// yarn rw info
+// ```
+let { cwd } = parser(hideBin(process.argv))
+cwd ??= process.env.RWJS_CWD
+const redwoodTomlPath = findup('redwood.toml', { cwd: cwd ?? process.cwd() })
 
-  try {
-    let cwd
-    if (argv.cwd) {
-      cwd = argv.cwd
-      // We delete the argument because it's not actually referenced in CLI,
-      // we use the `RWJS_CWD` env-var,
-      // and it conflicts with "forwarding" commands such as test and prisma.
-      delete argv.cwd
-    } else if (process.env.RWJS_CWD) {
-      cwd = process.env.RWJS_CWD
-    } else {
-      cwd = path.dirname(getConfigPath())
-    }
-
-    configPath = path.resolve(process.cwd(), cwd, 'redwood.toml')
-    if (!fs.existsSync(configPath)) {
-      throw new Error('Could not find `redwood.toml` config file.')
-    }
-
-    process.env.RWJS_CWD = cwd
-  } catch (e) {
-    console.error()
-    console.error('Error: Redwood CLI could not find your config file.')
-    console.error(`Expected '${configPath}'`)
-    console.error()
-    console.error(`Did you run Redwood CLI in a RedwoodJS project?`)
-    console.error(`Or specify an incorrect '--cwd' option?`)
-    console.error()
-    process.exit(1)
+try {
+  if (!redwoodTomlPath) {
+    throw new Error(
+      `Couldn't find a "redwood.toml" file--are you sure you're in a Redwood project?`
+    )
   }
+} catch (error) {
+  console.error(
+    [
+      `The Redwood CLI couldn't find your project's "redwood.toml".`,
+      'Did you run the Redwood CLI in a RedwoodJS project? Or specify "--cwd" incorrectly?',
+    ].join('\n')
+  )
+  process.exit(1)
 }
 
-const loadDotEnvDefaultsMiddleware = () => {
-  config({
-    path: path.join(getPaths().base, '.env'),
-    defaults: path.join(getPaths().base, '.env.defaults'),
-    multiline: true,
-  })
-}
+cwd ??= path.dirname(redwoodTomlPath)
+process.env.RWJS_CWD = cwd
 
-// eslint-disable-next-line no-unused-expressions
-yargs
+const rwPaths = getPaths()
+
+// # Load .env, .env.defaults
+//
+// This should be done as early as possible.
+// And the earliest we can do it is after setting the cwd.
+
+config({
+  path: path.join(rwPaths.base, '.env'),
+  defaults: path.join(rwPaths.base, '.env.defaults'),
+  multiline: true,
+})
+
+// # Build the CLI and run it
+
+yargs(hideBin(process.argv))
   .scriptName('rw')
   .middleware([
-    getCwdMiddleware,
-    loadDotEnvDefaultsMiddleware,
+    // We've already handled cwd above, but it's still in argv. Let's get rid of it.
+    (argv) => {
+      delete argv.cwd
+    },
     telemetryMiddleware,
   ])
   .option('cwd', {
     describe: 'Working directory to use (where `redwood.toml` is located.)',
```

Comment:
```suggestion
    describe: 'Working directory to use (where `redwood.toml` is located)',
```

If we actually wanted to keep the period it should be outside the last parenthesis